transactions/stakingCollection/register_multiple_nodes.cdc

```diff
@@ -0,0 +1,41 @@
+import FlowStakingCollection from 0xSTAKINGCOLLECTIONADDRESS
+
+/// Registers a delegator in the staking collection resource
+/// for the specified node information and the amount of tokens to commit
+
+transaction(ids: [String],
+            roles: [UInt8],
+            networkingAddresses: [String],
+            networkingKeys: [String],
+            stakingKeys: [String],
+            amounts: [UFix64],
+            publicKeys: [[String]?]) {
```

Comment:
why do we need `?` here, if we're always requiring at least 1 pub key?

Reply:
We don't require public keys if they are registering nodes that don't require machine accounts, so we allow them to make it `nil` here